internal/ingress/controller/parser/parser.go

```diff
@@ -223,9 +226,64 @@ func (p *Parser) Build() (*KongState, error) {
 		return nil, err
 	}
 
+	// populate CA certificates in Kong
+	state.CACertificates, err = p.getCACerts()
+
 	return &state, nil
 }
 
+func (p *Parser) getCACerts() ([]kong.CACertificate, error) {
+	caCertSecrets, err := p.store.ListCACerts()
+	if err != nil {
+		return nil, err
+	}
+
+	var caCerts []kong.CACertificate
+	for _, certSecret := range caCertSecrets {
+		secretName := certSecret.Namespace + "/" + certSecret.Name
+
+		idbytes, idExists := certSecret.Data["id"]
+		if !idExists {
+			glog.Errorf("invalid CA certificate in secret '%s':"+
+				" missing 'id' field in data", secretName)
+			continue
+		}
```

Comment:
I'm guessing the answer is probably "no, this isn't doable without considerable changes to what the controller maintains in state", but is there any way to handle this without specifying the ID and still map to KongPlugins? I think `Configuration` limits this also, for the same reason that using secret references for individual settings was difficult without breaking changes.

Assuming that is the case, we should probably have Kong Enterprise add an optional `name` field to CA certificates and name support in mTLS `ca_certificates` config to allow auto-generating the ID and using a human-friendly name in manifests. Requiring that users know they need to generate a UUID and set it is both a break from other CRDs and not immediately intuitive.